compiler/src/language_server/inlayhint.re

```diff
@@ -0,0 +1,89 @@
+open Grain;
+open Compile;
+open Grain_parsing;
+open Grain_utils;
+open Grain_typed;
+open Grain_diagnostics;
+open Sourcetree;
+open Lsp_types;
+
+// https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#inlayHintParams
+
+module RequestParams = {
+  [@deriving yojson({strict: false})]
+  type t = {
+    [@key "textDocument"]
+    text_document: Protocol.text_document_identifier,
+    range: Protocol.range,
+  };
+};
+
+// https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_inlayHint
+module ResponseResult = {
+  [@deriving yojson]
+  type inlay_hint = {
+    label: string,
+    position: Protocol.position,
+  };
+
+  [@deriving yojson]
+  type t = list(inlay_hint);
+};
+
+let send_no_result = (~id: Protocol.message_id) => {
+  Protocol.response(~id, `Null);
+};
+
+let find_hints = program => {
+  let hints = ref([]);
+  open Typedtree;
+  open Protocol;
+  module Iterator =
+    TypedtreeIter.MakeIterator({
+      include TypedtreeIter.DefaultIteratorArgument;
+      let enter_toplevel_stmt = (stmt: toplevel_stmt) => {
+        switch (stmt.ttop_desc) {
+        | TTopInclude(inc) =>
+          let path = inc.tinc_path;
+          let name =
+            switch (path) {
+            | PIdent(i) => i.name
+            | _ => ""
+            };
+
+          let stmt_loc = stmt.ttop_loc;
+          let stmt_end = stmt_loc.loc_end;
+
+          let p: Protocol.position = {
+            line: stmt_end.pos_lnum - 1,
+            character: stmt_end.pos_cnum - stmt_end.pos_bol + 1 + 1,
+          };
+
+          let r: ResponseResult.inlay_hint = {
+            label: " : " ++ name,
```

Comment:
Question on if the leading space should be removed making it `": " ++ name`, when you showed the picture in discord it looked a little off to me having the leading space.

Reply:
I think you are right for when it's in a hint.